pkg/network/ovn_kubernetes_dpu_host_test.go

```diff
@@ -0,0 +1,320 @@
+			},
+			forbiddenContainers: []string{},
+			expectedDaemonSet:   "ovnkube-node",
+		},
+		{
+			name:        "smart-nic mode",
+			ovnNodeMode: "smart-nic",
+			expectedContainers: []string{
+				"ovn-controller",
+				"ovn-acl-logging",
+				"kube-rbac-proxy-node",
+				"kube-rbac-proxy-ovn-metrics",
+				"northd",
+				"nbdb",
+				"sbdb",
+				"ovnkube-controller",
+			},
+			forbiddenContainers: []string{},
+			expectedDaemonSet:   "ovnkube-node-smart-nic",
+		},
+		{
+			name:        "dpu-host mode",
+			ovnNodeMode: "dpu-host",
+			expectedContainers: []string{
+				"kube-rbac-proxy-node",
+				"ovnkube-controller",
+			},
+			forbiddenContainers: []string{
+				"ovn-controller",
+				"ovn-acl-logging",
+				"kube-rbac-proxy-ovn-metrics",
+				"northd",
+				"nbdb",
+				"sbdb",
+			},
+			expectedDaemonSet: "ovnkube-node-dpu-host",
+		},
+	}
+
+	for _, template := range templates {
+		for _, mode := range modes {
+			testName := template.name + "_" + mode.name
+			t.Run(testName, func(t *testing.T) {
+				g := NewGomegaWithT(t)
+
+				// Create render data
+				data := createTestRenderData(mode.ovnNodeMode)
+
+				// Render the template
+				objs, err := render.RenderTemplate(template.templatePath, &data)
+				g.Expect(err).NotTo(HaveOccurred(), "Template rendering should succeed for %s %s", template.name, mode.name)
+				g.Expect(objs).To(HaveLen(1), "Should render exactly one object")
+
+				// Verify it's a DaemonSet with correct name
+				obj := objs[0]
+				g.Expect(obj.GetKind()).To(Equal("DaemonSet"))
+				g.Expect(obj.GetName()).To(Equal(mode.expectedDaemonSet))
+				g.Expect(obj.GetNamespace()).To(Equal("openshift-ovn-kubernetes"))
+
+				// Extract container names
+				containers, found, err := uns.NestedSlice(obj.Object, "spec", "template", "spec", "containers")
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(found).To(BeTrue())
+
+				var containerNames []string
+				for _, container := range containers {
+					cmap := container.(map[string]interface{})
+					name, found, err := uns.NestedString(cmap, "name")
+					g.Expect(err).NotTo(HaveOccurred())
+					g.Expect(found).To(BeTrue())
+					containerNames = append(containerNames, name)
+				}
+
+				// Verify expected containers are present
+				for _, expectedContainer := range mode.expectedContainers {
+					g.Expect(containerNames).To(ContainElement(expectedContainer),
+						"Container %s should be present in %s %s", expectedContainer, template.name, mode.name)
+				}
+
+				// Verify forbidden containers are absent
+				for _, forbiddenContainer := range mode.forbiddenContainers {
+					g.Expect(containerNames).NotTo(ContainElement(forbiddenContainer),
+						"Container %s should NOT be present in %s %s", forbiddenContainer, template.name, mode.name)
+				}
+			})
+		}
+	}
+}
+
+// TestOVNKubernetesNodeModeYAMLValidity tests that both templates generate valid YAML
+// for all OVN_NODE_MODE values
+func TestOVNKubernetesNodeModeYAMLValidity(t *testing.T) {
```

Comment:
Would it make sense to include this in the unit test above? In the end we're testing the same ovnk modes and rendering the yamls in both cases.